src/apiProvider.ts

```diff
@@ -16,16 +16,19 @@ const JSONbig = require("json-bigint");
 export class ApiProvider implements IApiProvider {
   private url: string;
   private timeoutLimit: number;
+  private config: AxiosRequestConfig;
 
   /**
    * Creates a new ApiProvider.
    * @param url the URL of the Elrond Api
    * @param timeout the timeout for any request-response, in milliseconds
+   * @param config axios request config options
    */
-  constructor(url: string, timeout?: number) {
+   constructor(url: string, timeout?: number, config?: AxiosRequestConfig) {
```

Comment:
Would it be a pain maybe if this won't be backwards compatible? The timeout param is in `AxiosRequestConfig`, I am thinking we should remove it from the constructor and use it in the newly config param.

Reply:
It should not be an issue for the way I'm using it for service. The problem is that this erdjs module is used heavily for all the web applications and I don't know that might be the implications to not keep it backwards compatible.